include/display.h

```diff
@@ -0,0 +1,312 @@
+/*
+ * Copyright (c) 2017 Jan Van Winkel <jan.van_winkel@dxplore.eu>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Public API for display drivers and applications
+ */
+
+#ifndef __DISPLAY_H__
+#define __DISPLAY_H__
+
+/**
+ * @brief Display Interface
+ * @defgroup display_interface display Interface
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#include <device.h>
+#include <stddef.h>
+#include <zephyr/types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum display_pixel_format {
+	PIXEL_FORMAT_RGB_888 = (1 << 0),
+};
+
+/**
+ * @struct display_capabilities
+ * @brief Structure holding display capabilities
+ *
+ * @var display_capabilities::x_resolution
+ * Display resolution in the X direction
+ *
+ * @var display_capabilities::y_resolution
+ * Display resolution in the Y direction
+ *
+ * @var display_capabilities::supported_pixel_formats
+ * Bitwise or of pixel formats supported by the display
+ *
+ * @var display_capabilities::current_pixel_format
+ * Currently active pixel format for the display
+ *
+ */
+struct display_capabilities {
+	u16_t x_resolution;
+	u16_t y_resolution;
+	u32_t supported_pixel_formats;
+	enum display_pixel_format current_pixel_format;
+};
+
+/**
+ * @typedef display_on_api
+ * @brief Callback API to turn display on
+ * See display_on() for argument description
+ */
+typedef int (*display_on_api)(const struct device *dev);
```

Comment:
I am not sure if  display_on/display_off are necessary because it can be done via Device Driver API.

Reply:
Do you mean the power management feature in the device driver API?
The intention of display_on/off API is to be used during normal operation.

Reply:
This could be useful incase you would like to refresh the full display (e.g. after rotation) over a slow serial interface (SPI/I2C). In this case you could turn off the display, refresh the screen and turn it back on without a visual tearing effect on the screen.